gluonnlp/data/sentiment.py

```diff
@@ -69,6 +70,202 @@ def _get_data(self):
             download(_get_repo_file_url('gluon/dataset/imdb', data_file_name),
                      path=root, sha1_hash=data_hash)
 
+    def _read_data(self):
+        with open(os.path.join(self._root, self._segment + '.json')) as f:
+            samples = json.load(f)
+        return samples
+
+
+class MR(SimpleDataset):
+    """Movie reviews for sentiment analysis.
+
+    From
+    https://www.cs.cornell.edu/people/pabo/movie-review-data/
+
+    Parameters
+    ----------
+    root : str, default '$MXNET_HOME/datasets/mr'
+        Path to temp folder for storing data.
+        MXNET_HOME defaults to '~/.mxnet'.
+    """
+    def __init__(self, root=os.path.join(_get_home_dir(), 'datasets', 'mr')):
+        self._data_file = {'all': ('all.json',
+                                   '7606efec578d9613f5c38bf2cef8d3e4e6575b2c')}
+        root = os.path.expanduser(root)
+        if not os.path.isdir(root):
+            os.makedirs(root)
+        self._root = root
+        self._get_data()
+        super(MR, self).__init__(self._read_data())
+
+    def _get_data(self):
+        data_file_name, data_hash = self._data_file['all']
+        root = self._root
+        path = os.path.join(root, data_file_name)
+        if not os.path.exists(path) or not check_sha1(path, data_hash):
+            download(_get_repo_file_url('gluon/dataset/mr', data_file_name),
+                     path=root, sha1_hash=data_hash)
+
+    def _read_data(self):
```

Comment:
The `_read_data` and `_get_data` have opportunities for refactoring. Given the number of datasets that share the same logic, we may consider introducing a base class that provides `_get_data` and `_read_data`.

Reply:
resolved.